crates/oauth2-types/src/scope.rs

```diff
@@ -14,35 +14,86 @@
 
 #![allow(clippy::module_name_repetitions)]
 
-use std::{borrow::Cow, collections::HashSet, iter::FromIterator, ops::Deref, str::FromStr};
+use std::{collections::HashSet, fmt, iter::FromIterator, str::FromStr};
 
 use itertools::Itertools;
 use serde::{Deserialize, Serialize};
+use serde_with::{DeserializeFromStr, SerializeDisplay};
 use thiserror::Error;
 
 #[derive(Debug, Error, PartialEq, Eq, PartialOrd, Ord, Hash)]
 #[error("Invalid scope format")]
 pub struct InvalidScope;
 
-#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub struct ScopeToken(Cow<'static, str>);
+/// Tokens to define the scope of an access token or to request specific claims.
+#[derive(
+    Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, SerializeDisplay, DeserializeFromStr,
+)]
+pub enum ScopeToken {
```

Comment:
I don't think that should live here. This crate is supposed to be usable in non-Matrix context, so having the Matrix device scope as a enum member here feels weird.

I'd rather keep the `ScopeToken` here as is, and have this type live somewhere else, with an `impl From<ScopeToken> for WhateverScopeType` and `impl TryFrom<WhateverScopeType> for ScopeToken`. What do you think?

Reply:
Sure, should it just be a new type in the client crate?